client/app/components/api/sections/documents/DocumentView.jsx

```diff
@@ -0,0 +1,57 @@
+var React = require('react');
+var Router = require('react-router');
+var Link = Router.Link;
+var RouteHandler = Router.RouteHandler;
+var Markdown = require('react-remarkable');
+
+var fetchDocument = function (data, callback) {
+  if (data.section === "components"){
+    if (data.document === "form"){
+      callback("Hello World -FORM");
+    }
+    if (data.document === "router"){
+      callback("Hello World -router");
+    }
+  }
+}
+
+var DocumentView = React.createClass({
+  mixins: [ Router.State ],
+  getInitialState: function (){
+    return {
+      documentToRender: null
+    };
+  },
+  getDocumentsFromStore: function () {
+    var section = this.getParams().section;
+    var document = this.getParams().document;
+    fetchDocument({
+      section: section,
+      document: document
+    },function (documentToRender) {
+      this.setState({documentToRender: documentToRender});
+    }.bind(this));
+  },
+  componentWillReceiveProps: function(){
```

Comment:
Is there a particular reason for having this componentWillReceiveProps function here?